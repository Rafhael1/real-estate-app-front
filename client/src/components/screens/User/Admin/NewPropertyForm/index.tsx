import React, { useState } from "react"

import { useSelector, useDispatch } from "react-redux"

import { RootStore } from "../../../../../redux/store"

import axios from "../../../../../utils/api/axios"

const NewPropertyForm = () => {

	const [propertyInfo, setPropertyInfo] = useState({
		title: "",
		description: "",
		address: "",
		country: "",
		price: "",
		status: "",
	})

	const [image1, setImage1] = useState(null)
	const [image2, setImage2] = useState(null)
	const [image3, setImage3] = useState(null)
	const [image4, setImage4] = useState(null)
	const [image5, setImage5] = useState(null)

	const {
		title,
		description,
		address,
		country,
		price,
		status,
	} = propertyInfo

	const onChangeHandler = (e: any) => {
		setPropertyInfo({
			...propertyInfo,
			[e.target.name]: e.target.value 
		})
	}

	const onSubmitHandler = async (e: any) => {
		e.preventDefault()
		try {
			const formData: any = new FormData()
			formData.append("images", image1)
			formData.append("images", image2)
			formData.append("images", image3)
			formData.append("images", image4)
			formData.append("images", image5)

			const obj = {
				hello: "world"
			}
			const json = JSON.stringify(obj)
			formData.append("body", json)
    
			const res = await axios.post("/upload", formData) 
    
			console.log(res)
    
		} catch (error) {
			console.log(error)
		}
	}

	return (
		<form onSubmit={(e) => onSubmitHandler(e)} >
			<input type="file" name="images" onChange={(e: any) => setImage1(e.target.files[0])} />
			<input type="file" name="images" onChange={(e: any) => setImage2(e.target.files[0])} />
			<input type="file" name="images" onChange={(e: any) => setImage3(e.target.files[0])} />
			<input type="file" name="images" onChange={(e: any) => setImage4(e.target.files[0])} />
			<input type="file" name="images" onChange={(e: any) => setImage5(e.target.files[0])} />
			<button type="submit" >Submit</button>
		</form>
	)
}

export default NewPropertyForm