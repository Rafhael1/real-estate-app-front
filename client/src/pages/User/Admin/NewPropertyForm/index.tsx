import React, { useState } from "react"

import { useSelector, useDispatch } from "react-redux"

import { RootStore } from "../../../../redux/store"

import { Form, Field, reduxForm } from "redux-form"

import axios from "../../../../components/utils/api/axios"

const NewPropertyForm = () => {

	const [propertyInfo, setPropertyInfo] = useState({
		title: "",
		description: "",
		address: "",
		country: "",
		price: "",
		status: "",
	})

	const [image1, setImage1] = useState(null)
	const [image2, setImage2] = useState(null)
	const [image3, setImage3] = useState(null)
	const [image4, setImage4] = useState(null)
	const [image5, setImage5] = useState(null)

	const {
		title,
		description,
		address,
		country,
		price,
		status,
	} = propertyInfo

	const onChangeHandler = (e: any) => {
		setPropertyInfo({
			...propertyInfo,
			[e.target.name]: e.target.value 
		})
	}

	const onSubmitHandler = async (e: any) => {
		e.preventDefault()
		try {
			const formData: any = new FormData()
			formData.append("images", image1)
			formData.append("images", image2)
			formData.append("images", image3)
			formData.append("images", image4)
			formData.append("images", image5)

			const obj = {
				hello: "world"
			}
			const json = JSON.stringify(obj)
			formData.append("body", json)
    
			const res = await axios.post("/upload", formData) 
    
			console.log(res)
    
		} catch (error) {
			console.log(error)
		}
	}

	return (
		<Form onSubmit={(e) => onSubmitHandler(e)} >
			<Field name="title" type="text" />
			<Field name="description" type="text" />
			<Field name="address" type="text" />
			<Field name="country" type="text" />
			<Field name="price" type="text" />
			<Field name="status" type="text" />

			<Field name="images" type="file" />
			<Field name="images" type="file" />
			<Field name="images" type="file" />
			<Field name="images" type="file" />
			<Field name="images" type="file" />
			<button type="submit" >Submit</button>
		</Form>
	)
}

export default reduxForm({
	form: "NewPropertyForm"
})(NewPropertyForm)