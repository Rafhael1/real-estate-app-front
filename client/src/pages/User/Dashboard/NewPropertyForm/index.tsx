import React from "react"

import { useDispatch } from "react-redux"


import { Form, Field, reduxForm } from "redux-form"

import { addNewRealEstate } from "../../../../redux/actions/RealEstates"

import renderFileInput from "components/fields/inputFileField"

const NewPropertyForm = ({ handleSubmit }: any): JSX.Element => {

	// const [propertyInfo, setPropertyInfo] = useState({
	// 	title: "",
	// 	description: "",
	// 	address: "",
	// 	country: "",
	// 	price: "",
	// 	status: "",
	// })

	// const [image1, setImage1] = useState(null)
	// const [image2, setImage2] = useState(null)
	// const [image3, setImage3] = useState(null)
	// const [image4, setImage4] = useState(null)
	// const [image5, setImage5] = useState(null)

	// const {
	// 	title,
	// 	description,
	// 	address,
	// 	country,
	// 	price,
	// 	status,
	// } = propertyInfo

	// const onChangeHandler = (e: any) => {
	// 	setPropertyInfo({
	// 		...propertyInfo,
	// 		[e.target.name]: e.target.value 
	// 	})
	// }

	// const onSubmitHandler = async (e: any) => {
	// 	e.preventDefault()
	// 	try {
	// 		const formData: any = new FormData()
	// 		formData.append("images", image1)
	// 		formData.append("images", image2)
	// 		formData.append("images", image3)
	// 		formData.append("images", image4)
	// 		formData.append("images", image5)

	// 		const obj = {
	// 			hello: "world"
	// 		}
	// 		const json = JSON.stringify(obj)
	// 		formData.append("body", json)
    
	// 		const res = await axios.post("/upload", formData) 
    
	// 		console.log(res)
    
	// 	} catch (error) {
	// 		console.log(error)
	// 	}
	// }

	const dispatch = useDispatch()

	return (
		<Form onSubmit={handleSubmit((values: any) => dispatch(addNewRealEstate(values)))} >
			<Field name="title" component="input"  type="text" />
			<Field name="description" component="input"  type="text" />
			<Field name="squareMeter" component="input"  type="number" />
			<Field name="bathrooms" component="input"  type="number" />
			<Field name="bedrooms" component="input"  type="number" />
			<Field name="address" component="input"  type="text" />
			<Field name="country" component="input"  type="text" />
			<Field name="price" component="input"  type="text" />
			<Field name="status" component="input"  type="text" />
			<Field name="image1" component={renderFileInput}  />
			<Field name="image2" component="input"  type="file" />
			<Field name="image3" component="input"  type="file" />
			<Field name="image4" component="input"  type="file" />
			<Field name="image5" component="input"  type="file" />
			<button type="submit" >Submit</button>
		</Form>
	)
}

export default reduxForm({
	form: "NewPropertyFormRedux"
})(NewPropertyForm)