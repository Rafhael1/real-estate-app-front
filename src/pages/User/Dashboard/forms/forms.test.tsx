/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import { shallow } from 'enzyme'
import NewPropertyForm from './newPropertyForm'


const props: any = {
  handleSubmit: () => {},
}

it('Should render new property form', () => {
  expect(shallow(<NewPropertyForm {...props} />)).toMatchSnapshot()
})