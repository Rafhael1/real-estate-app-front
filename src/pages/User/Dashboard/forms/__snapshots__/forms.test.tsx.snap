// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render new property form 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="NewPropertyFormRedux"
  forwardedRef={null}
  getFormState={[Function]}
  handleSubmit={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
